jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java

```diff
@@ -14,4 +20,195 @@ public class JdbcTemplate {
+            }
+        }, args);
+    }
+
+    public <T> List<T> query(String sql, RowMapper<T> rm) {
+        return context(new PreparedStrategy() {
+            @Override
+            public PreparedStatement createStatement(Connection connection) throws SQLException {
+                return connection.prepareStatement(sql);
+            }
+        }, rm);
+    }
+
+    public <T> T queryForObject(String sql, RowMapper<T> rm, Object... args) {
+        List<T> list = context(new PreparedStrategy() {
+            @Override
+            public PreparedStatement createStatement(Connection connection) throws SQLException {
+                return connection.prepareStatement(sql);
+            }
+        }, rm, args);
+
+        if (list.isEmpty()) {
+            return null;
+        }
+        return list.get(0);
+    }
+
+    public void context(PreparedStrategy preparedStrategy) {
+        Connection conn = null;
+        PreparedStatement pstmt = null;
+        try {
+            conn = dataSource.getConnection();
+            pstmt = preparedStrategy.createStatement(conn);
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        } finally {
+            try {
+                if (pstmt != null) {
+                    pstmt.close();
+                }
+            } catch (SQLException ignored) {
+            }
+
+            try {
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException ignored) {
+            }
+        }
+    }
+
+    public void context(PreparedStrategy preparedStrategy, Object[] args) {
+        Connection conn = null;
+        PreparedStatement pstmt = null;
+        try {
+            conn = dataSource.getConnection();
+            pstmt = preparedStrategy.createStatement(conn);
+
+            for (int i = 0; i < args.length; i++) {
+                pstmt.setObject(i + 1, args[i]);
+            }
+
+            pstmt.executeUpdate();
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        } finally {
+            try {
+                if (pstmt != null) {
+                    pstmt.close();
+                }
+            } catch (SQLException ignored) {
+            }
+
+            try {
+                if (conn != null) {
+                    conn.close();
+                }
+            } catch (SQLException ignored) {
+            }
+        }
+    }
+
+    public <T> List<T> context(PreparedStrategy preparedStrategy, RowMapper<T> rm) {
+        Connection conn = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            conn = dataSource.getConnection();
+            pstmt = preparedStrategy.createStatement(conn);
+            rs = pstmt.executeQuery();
+
+            ArrayList<T> list = new ArrayList<>();
```

Comment:
```suggestion
            List<T> list = new ArrayList<>();
```
인터페이스에 대고 프로그래밍하기 ! 에 대해서는 어떻게 생각하시나용 🧐

Reply:
오 넘 좋습니다!!